### PR TITLE
Agents see an appointment preview before they can confirm

### PR DESCRIPTION
**Pull request by AndrewVos**
~~@benbarnett Not sure about that copy, so keeping in `[draft]` for now.~~

~~This is my third approach at doing this. The other two were quite ugly and used javascript :cry:~~

~~I'm sure there are some nice refactorings that could be done so I'm taking a look at that now. Feel free to review it now though.~~

image

### CODE REVIEW

File: app/controllers/appointments_controller.rb [hunk 1]
```diff
@@ -50,10 +50,21 @@ def new
...
+      render :new
+    end
+  end
+
   def create
     @appointment = Appointment.new(create_params.merge(agent: current_user))
     @appointment.assign_to_guider
-    if @appointment.save
+
+    if !edit_appointment? && @appointment.save
```
Comment (benlovell) [inline app/controllers/appointments_controller.rb]: Can we remove the negation? If we switch the helper to something like this instead:

```ruby
def creating?
  params[:edit_appointment]
end
```

File: app/views/appointments/preview.html.erb [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    { title: 'Preview' },
+  )
+%>
+
+<h1>Preview appointment</h1>
+
+<%= rebooked_from_heading(@appointment) %>
+
+<div class="alert alert-warning">
+  This booking has not been created yet. Please confirm the details are correct before proceeding.
```
Comment (benlovell) [inline app/views/appointments/preview.html.erb]: We should say 'appointment' not 'booking' for consistency.

File: app/views/appointments/preview.html.erb [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  <%= f.hidden_field :last_name %>
+  <%= f.hidden_field :email %>
+  <%= f.hidden_field :phone %>
+  <%= f.hidden_field :mobile %>
+  <%= f.hidden_field :memorable_word %>
+  <%= f.hidden_field :notes %>
+  <%= f.hidden_field :opt_out_of_market_research %>
+  <%= f.hidden_field :status %>
+
+  <div class='t-preview'>
```
Comment (Guntrisoft) [inline app/views/appointments/preview.html.erb]: quotes

File: spec/features/agent_manages_appointments_spec.rb [hunk 4]
```diff
@@ -190,6 +221,59 @@ def when_they_want_to_book_an_appointment
...
+  def when_they_go_back_to_edit_their_appointment
+    @page = Pages::PreviewAppointment.new
+    expect(@page).to be_displayed
+    @page.edit_appointment.click
+  end
+
+  def and_they_change_some_details
+    @page = Pages::NewAppointment.new
+
+    @page.first_name.set                 'Another'
```
Comment (benlovell) [inline spec/features/agent_manages_appointments_spec.rb]: Not a huge fan of this alignment, to me at least it's actually more difficult to scan / read. Do you find it much easier? (not a blocker, just a question)

**Comment by Guntrisoft** [comment]
Markup wise I think it's ok, although I might have gone with bold label's for each item

**Comment by AndrewVos** [comment]
@Guntrisoft Changed to `<strong>`

**Pull request merged**